sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -84,24 +93,43 @@ case class CreateTableLikeCommand(
       provider
     } else if (sourceTableDesc.tableType == CatalogTableType.VIEW) {
       Some(sparkSession.sessionState.conf.defaultDataSourceName)
+    } else if (fileFormat.inputFormat.isDefined) {
+      Some(DDLUtils.HIVE_PROVIDER)
     } else {
       sourceTableDesc.provider
     }
 
+    val newStorage = if (fileFormat.inputFormat.isDefined) {
+      sourceTableDesc.storage.copy(
+        locationUri = fileFormat.locationUri,
+        inputFormat = fileFormat.inputFormat,
+        outputFormat = fileFormat.outputFormat,
+        serde = fileFormat.serde,
+        properties = fileFormat.properties,
+        compressed = sourceTableDesc.storage.compressed
```

Comment:
does it make sense to retain the "compressed" flag from source table? I think it's more reasonable to use everything from what users specified, for the serde.